src/directory/managed_directory.rs

```diff
@@ -267,9 +264,7 @@ impl Directory for ManagedDirectory {
         let read_only_source = self.directory.open_read(path)?;
         let (footer, reader) = Footer::extract_footer(read_only_source)
             .map_err(|err| IOError::with_path(path.to_path_buf(), err))?;
-        if !footer.is_compatible() {
-            return Err(OpenReadError::IncompatibleIndex(footer));
-        }
+        footer.is_compatible().unwrap();
```

Comment:
Since I deleted the `From<TantivyError> for OpenReadError` shim, returning the Incompatibility wrapped inside the Err() from is_compatible makes it impossible to propagate it further. 

What do you want to return here?